codegenerator/cli/templates/static/codegen/src/eventFetching/ChainManager.res

```diff
@@ -72,12 +72,35 @@ type fetchStateWithData = {
   heighestBlockBelowThreshold: int,
 }
 
+let isQueueItemEarlierUnorderedBelowReorgThreshold = (
+  ~fetchStatesMap: ChainMap.t<fetchStateWithData>,
+) => (a: multiChainEventComparitor, b: multiChainEventComparitor) => {
+  let isItemBelowReorgThreshold = item => {
+    item.earliestEvent->FetchState.queueItemIsInReorgThreshold(
+      ~heighestBlockBelowThreshold=ChainMap.get(
+        fetchStatesMap,
+        a.chain,
+      ).heighestBlockBelowThreshold,
+    )
+  }
+  switch (a->isItemBelowReorgThreshold, b->isItemBelowReorgThreshold) {
+  | (false, true) => true
+  | (true, false) => false
+  | _ => isQueueItemEarlierUnorderedMultichain(a, b)
+  }
+}
+
 let determineNextEvent = (
   fetchStatesMap: ChainMap.t<fetchStateWithData>,
   ~isUnorderedMultichainMode: bool,
+  ~onlyBelowReorgThreshold: bool,
 ): result<isInReorgThresholdRes<multiChainEventComparitor>, noItemsInArray> => {
   let comparitorFunction = if isUnorderedMultichainMode {
-    chainFetcherPeekComparitorEarliestEventPrioritizeEvents
+    if onlyBelowReorgThreshold {
+      isQueueItemEarlierUnorderedBelowReorgThreshold(~fetchStatesMap)
+    } else {
+      isQueueItemEarlierUnorderedMultichain
+    }
   } else {
     isQueueItemEarlier
   }
```

Comment:
Could we not pass this comparative function in as an argument rather than passing in and checking these booleans on each call?